modules/unsupported/process-feature/src/test/java/org/geotools/process/vector/ClassifyByRangeProcessTest.java

```diff
@@ -0,0 +1,205 @@
+        assertEquals(String.class, classAttribute.getType().getBinding());
+    }
+    
+    @Test
+    public void testClassifyDefaultOutputValues() {
+        ClassifyByRangeProcess cp = new ClassifyByRangeProcess();
+        SimpleFeatureCollection result = cp.execute(fc, "value", "2,5,10", null, null, false, "class", null, null);
+        assertEquals(3, result.size());
+        SimpleFeatureIterator iterator = result.features();
+        SimpleFeature ft = iterator.next();
+        assertEquals("1", ft.getAttribute("class"));
+        ft = iterator.next();
+        assertEquals("3", ft.getAttribute("class"));
+    }
+    
+    @Test
+    public void testClassifyWrongValues() {
+        ClassifyByRangeProcess cp = new ClassifyByRangeProcess();
+        boolean error = false;
+        try {
+            cp.execute(fc, "value", "2,5,10", null, null, true, "class", "A,B,C,D,E", null);
+        } catch(ProcessException e) {
+            error = true;
+        }
+        assertTrue(error);
+    }
+    
+    @Test
+    public void testClassifyWithClassifierEqualInterval() {
+        ClassifyByRangeProcess cp = new ClassifyByRangeProcess();
+        SimpleFeatureCollection result = cp.execute(fc, "value", null, "EqualInterval", 3, true, "class", "A,B,C,D,E", null);
+        assertEquals(3, result.size());
+        SimpleFeatureIterator iterator = result.features();
+        SimpleFeature ft = iterator.next();
+        assertEquals("A", ft.getAttribute("class"));
+        ft = iterator.next();
+        assertEquals("C", ft.getAttribute("class"));
+        ft = iterator.next();
+        assertEquals("D", ft.getAttribute("class"));
+    }
+    
+    @Test
+    public void testClassifyWithClassifierEqualIntervalMoreClasses() {
+        ClassifyByRangeProcess cp = new ClassifyByRangeProcess();
+        SimpleFeatureCollection result = cp.execute(fc, "value", null, "EqualInterval", 5, true, "class", "A,B,C,D,E,F,G", null);
+        assertEquals(3, result.size());
+        SimpleFeatureIterator iterator = result.features();
+        SimpleFeature ft = iterator.next();
+        assertEquals("A", ft.getAttribute("class"));
+        ft = iterator.next();
+        assertEquals("D", ft.getAttribute("class"));
+        ft = iterator.next();
+        assertEquals("F", ft.getAttribute("class"));
+    }
+    
+    @Test
+    public void testClassifyWithQuantileInterval() {
+        ClassifyByRangeProcess cp = new ClassifyByRangeProcess();
+        SimpleFeatureCollection result = cp.execute(fc, "value", null, "Quantile", 3, true, "class", "A,B,C,D,E", null);
+        assertEquals(3, result.size());
+        SimpleFeatureIterator iterator = result.features();
+        SimpleFeature ft = iterator.next();
+        assertEquals("A", ft.getAttribute("class"));
+        ft = iterator.next();
+        assertEquals("B", ft.getAttribute("class"));
+        ft = iterator.next();
+        assertEquals("C", ft.getAttribute("class"));
+    }
+}
```

Comment:
Maybe add a test for invalid classifier function, in case using an enum for it is not suitable